File: LENGUAJES/PHP/Ejercicio8.php
<?php
// INTERACCION USUARIO

function obtenerAdivinanza() {
    while (true) {
        echo "Adivina el número del 1 al 10: ";
        $input = trim(fgets(STDIN));
        
        if (is_numeric($input) && intval($input) >= 1 && intval($input) <= 10) {
            return intval($input);
        } else {
            echo "Por favor, ingresa un número entre 1 y 10.\n";
        }
    }
}

function jugar() {
    $num = rand(1, 10);
    $intentos = 3; // Número de intentos permitidos

    echo "Tienes $intentos intentos para adivinar el número.\n";
    
    for ($i = 0; $i < $intentos; $i++) {
        $adivinar = obtenerAdivinanza();
        
        if ($num === $adivinar) {
            echo "¡Has ganado!\n";
            return;
        } else {
            echo "Intento " . ($i + 1) . " fallido.\n";
        }
    }
    
    echo "Has perdido, el número era: $num\n";
}

function main() {
    while (true) {
        jugar();
        echo "¿Quieres jugar de nuevo? (s/n): ";
        $continuar = trim(fgets(STDIN));
        
        if (strtolower($continuar) !== 's') {
            echo "Gracias por jugar. ¡Hasta luego!\n";
            break;
        }
    }
}
main();
?>

File: LENGUAJES/PYTHON/Ejercicio8.py
#Interaccion Usuario

import random

def obtener_adivinanza():
    while True:
        try:
            adivinar = int(input("Adivina el número del 1 al 10: "))
            if 1 <= adivinar <= 10:
                return adivinar
            else:
                print("Por favor, ingresa un número entre 1 y 10.")
        except ValueError:
            print("Error: Debes ingresar un número entero.")

def jugar():
    num = random.randint(1, 10)
    intentos = 3  # Número de intentos permitidos

    print("Tienes 3 intentos para adivinar el número.")
    
    for i in range(intentos):
        adivinar = obtener_adivinanza()
        
        if num == adivinar:
            print("¡Has ganado!")
            return
        else:
            print(f"Intento {i + 1} fallido.")
    
    print(f"Has perdido, el número era: {num}")

def main():
    while True:
        jugar()
        continuar = input("¿Quieres jugar de nuevo? (s/n): ").strip().lower()
        if continuar != 's':
            print("Gracias por jugar. ¡Hasta luego!")
            break
main()